openmc/mgxs_library.py

```diff
@@ -0,0 +1,2577 @@
+        return self._atomic_weight_ratio
+
+    @property
+    def fissionable(self):
+        return self._fissionable
+
+    @property
+    def temperatures(self):
+        return self._temperatures
+
+    @property
+    def scatter_format(self):
+        return self._scatter_format
+
+    @property
+    def order(self):
+        return self._order
+
+    @property
+    def num_polar(self):
+        return self._num_polar
+
+    @property
+    def num_azimuthal(self):
+        return self._num_azimuthal
+
+    @property
+    def total(self):
+        return self._total
+
+    @property
+    def absorption(self):
+        return self._absorption
+
+    @property
+    def scatter_matrix(self):
+        return self._scatter_matrix
+
+    @property
+    def multiplicity_matrix(self):
+        return self._multiplicity_matrix
+
+    @property
+    def fission(self):
+        return self._fission
+
+    @property
+    def nu_fission(self):
+        return self._nu_fission
+
+    @property
+    def prompt_nu_fission(self):
+        return self._prompt_nu_fission
+
+    @property
+    def delayed_nu_fission(self):
+        return self._delayed_nu_fission
+
+    @property
+    def kappa_fission(self):
+        return self._kappa_fission
+
+    @property
+    def chi(self):
+        return self._chi
+
+    @property
+    def chi_prompt(self):
+        return self._chi_prompt
+
+    @property
+    def chi_delayed(self):
+        return self._chi_delayed
+
+    @property
+    def num_orders(self):
+        if self._order is not None:
+            if self._scatter_format in (None, 'legendre'):
+                return self._order + 1
+            else:
+                return self._order
+
+    @property
+    def xs_shapes(self):
+
+        if self._xs_shapes is None:
+
+            self._xs_shapes = {}
+            self._xs_shapes["[G]"] = (self.energy_groups.num_groups,)
+            self._xs_shapes["[G']"] = (self.energy_groups.num_groups,)
+            self._xs_shapes["[G][G']"] = (self.energy_groups.num_groups,
+                                          self.energy_groups.num_groups)
+            self._xs_shapes["[DG]"] = (self.num_delayed_groups,)
+            self._xs_shapes["[DG][G]"] = (self.num_delayed_groups,
+                                          self.energy_groups.num_groups)
+            self._xs_shapes["[DG][G']"] = (self.num_delayed_groups,
+                                           self.energy_groups.num_groups)
+            self._xs_shapes["[DG][G][G']"] = (self.num_delayed_groups,
+                                              self.energy_groups.num_groups,
+                                              self.energy_groups.num_groups)
+
+            self._xs_shapes["[G][G'][Order]"] \
+                = (self.energy_groups.num_groups,
+                   self.energy_groups.num_groups, self.num_orders)
+
+            # If representation is by angle prepend num polar and num azim
+            if self.representation == 'angle':
```

Comment:
This is an example of where using the `ANGLE` constant would is preferable to writing out the string explicitly. Doing so reduces bugs related to typos ('angels') and constants make it easy for your coding editor/environment to autocomplete the values.